include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -0,0 +1,683 @@
+       * has changed, this can also reconfigured by passing in the appropriate
+       * arguments to the function.
+       *
+       * @note This also resets the active tape number to an invalid number, and
+       * deactivates the recording mode for taped variables.
+       */
+      virtual void
+      reset(const unsigned int n_independent_variables = 0,
+            const unsigned int n_dependent_variables   = 0,
+            const bool         clear_registered_tapes  = true);
+
+      /**
+       * Returns whether or not this class is tracking calculations performed
+       * with its marked independent variables.
+       */
+      bool
+      is_recording() const;
+
+      /**
+       * Returns the tape number which is currently activated for recording or
+       * reading.
+       */
+      types::tape_index
+      active_tape() const;
+
+      /**
+       * Returns whether or not a tape number has already been used
+       * or registered.
+       */
+      bool
+      is_registered_tape(const types::tape_index &tape_index) const;
+
+      /**
+       * Set the buffer sizes for the next active tape.
+       *
+       * This function must be called before enable_record_sensitivities() for
+       * it to have any influence on the memory allocated to the next recorded
+       * tape.
+       *
+       * @note This only has an effect when using ADOL-C numbers. As stated by
+       * the ADOL-C manual, it may be desirable to create a file ".adolcrc" in
+       * the program run directory and set the buffer size therein.
+       * Alternatively, this function can be used to override the settings for
+       * any given tape, or can be used in the event that no ".adolcrc" file
+       * exists.
+       *
+       * @param[in] obufsize ADOL-C operations buffer size
+       * @param[in] lbufsize ADOL-C locations buffer size
+       * @param[in] vbufsize ADOL-C value buffer size
+       * @param[in] tbufsize ADOL-C Taylor buffer size
+       */
+      void
+      set_tape_buffer_sizes(const types::tape_buffer_sizes obufsize = 67108864,
+                            const types::tape_buffer_sizes lbufsize = 67108864,
+                            const types::tape_buffer_sizes vbufsize = 67108864,
+                            const types::tape_buffer_sizes tbufsize = 67108864);
+
+      /**
+       * Enable recording mode for a given tape. The use of this function is
+       * mandatory if the auto-differentiable number is a taped type. However,
+       * for the purpose of developing generic code, it can also be safely
+       * called for tapeless auto-differentiable numbers.
+       *
+       * The operations that take place between this function call and that
+       * of disable_record_sensitivities() are recorded to the tape and can
+       * be replayed and reevaluated as necessary.
+       *
+       * The typical set of operations to be performed during this phase are:
```

Comment:
with "this phase", do you mean "the phase between enable_...() and disable_...() calls"? If so maybe make this clearer. For example, you could give this phase a name in the previous paragraph and reference it here.